enos/modules/artifact/metadata/main.tf

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+// Given the architecture, distro, version, edition, and desired package type,
+// return the metadata for an artifact.
+
+variable "arch" {
+  description = "The artifact platform architecture"
+  type        = string
+
+  validation {
+    condition     = contains(["amd64", "arm64", "s390x"], var.arch)
+    error_message = <<-EOF
+      distro must be one of "amd64", "arm64", "s390x"
+    EOF
+  }
+}
+
+variable "distro" {
+  description = "The target operating system distro"
+  type        = string
+
+  validation {
+    condition     = contains(["amzn", "leap", "rhel", "sles", "ubuntu"], var.distro)
+    error_message = <<-EOF
+      distro must be one of "amzn", "leap", "rhel", "sles", "ubuntu"
+    EOF
+  }
+}
+
+variable "edition" {
+  description = "The Vault edition. E.g. ent or ent.hsm.fips1403"
+  type        = string
+
+  validation {
+    condition     = contains(["oss", "ce", "ent", "ent.fips1402", "ent.fips1403", "ent.hsm", "ent.hsm.fips1402", "ent.hsm.fips1403"], var.edition)
+    error_message = <<-EOF
+      edition must be one of "oss", "ce", "ent", "ent.fips1402", "ent.fips1403", "ent.hsm", "ent.hsm.fips1402", "ent.hsm.fips1403"
+    EOF
+  }
+}
+
+variable "package_type" {
+  description = "The artifact packaging type"
+  type        = string
+
+  validation {
+    condition     = contains(["package", "rpm", "deb", "zip", "bundle"], var.package_type)
+    error_message = <<-EOF
+      package_type must be one of "package", "rpm", "deb", "zip", "bundle"
+    EOF
+  }
+}
+
+variable "vault_version" {
```

Comment:
It would be great if we could run scenarios for all supported Vault versions. I don't think we have a way to dynamically get that information though. So we may be left hardcoding the versions for now.